### PR TITLE
fix: makefile on macOS

### LINKED ISSUE #2438: Makefile doesn't support macOS now
LLVM auto detect doesn't work on macOS now, because homebrew installed llvm is key-only:
```
$ brew info llvm
llvm: stable 13.0.0 (bottled), HEAD [keg-only]
...
==> Caveats
To use the bundled libc++ please add the following LDFLAGS:
  LDFLAGS="-L/opt/homebrew/opt/llvm/lib -Wl,-rpath,/opt/homebrew/opt/llvm/lib"

llvm is keg-only, which means it was not symlinked into /opt/homebrew,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.
```

To workaround this, we need to modify this `Makefile` and manually specify the correct path for llvm on mac. 
For arm64(M1):
```
CLANG ?= /opt/homebrew/Cellar/llvm@11/11.1.0_3/bin/clang
LLVM_AR ?= /opt/homebrew/Cellar/llvm@11/11.1.0_3/bin/llvm-ar
LLVM_NM ?= /opt/homebrew/Cellar/llvm@11/11.1.0_3/bin/llvm-nm
```

For amd64(x86_64):
```
CLANG ?= /usr/local/Cellar/llvm@11/11.1.0_3/bin/clang
LLVM_AR ?= /usr/local/Cellar/llvm@11/11.1.0_3/bin/llvm-ar
LLVM_NM ?= /usr/local/Cellar/llvm@11/11.1.0_3/bin/llvm-nm
```

### PR DESCRIPTION
**Pull request by PureWhiteWu**
fixes #2438.

### CODE REVIEW

**Comment by deadprogram** [comment]
@dgryski @dkegel-fastly can you please try this out on your Mac setups?

**Comment by ysoldak** [comment]
Works for me on Intel Mac, Monterey (12.1)

File: Makefile [hunk 1]
```diff
@@ -7,13 +7,14 @@ LLVM_BUILDDIR ?= llvm-build
 LLVM_PROJECTDIR ?= llvm-project
 CLANG_SRC ?= $(LLVM_PROJECTDIR)/clang
 LLD_SRC ?= $(LLVM_PROJECTDIR)/lld
+BREW_PREFIX := $(shell brew --prefix)
 
 # Try to autodetect LLVM build tools.
 # FIXME(#2438): This doesn't work on darwin now. Homebrew installed llvm is key-only.
```
Comment (dkegel-fastly) [inline Makefile]: do you want to remove the FIXME?

Comment (deadprogram) [reply Makefile]: Oops! Yes that needs to be removed.

Comment (deadprogram) [reply Makefile]: Actually I fixed it.

**Pull request merged**